llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2952,5 +2952,22 @@ RISCVTTIImpl::enableMemCmpExpansion(bool OptSize, bool IsZeroCmp) const {
     Options.LoadSizes = {4, 2, 1};
     Options.AllowedTailExpansions = {3};
   }
+
+  if (IsZeroCmp && ST->hasVInstructions() && ST->enableUnalignedVectorMem()) {
```

Comment:
Do we still need the enableUnalignedVectorMem check? If I'm understanding this right MemcmpExpand will generate a scalar which should be ok because we check for enableUnalignedScalarMem. 

And then in the new combine we're not actually changing the load at all. There must be some other existing combine which is converting the scalar load to a vector load, which should be respecting alignment?